idx/idx-api.php

```diff
@@ -768,17 +768,94 @@ public function get_leads( $timeframe = null, $start_date = '' ) {
 		}
 
 		$lead_data = $this->idx_api( $api_method . $offset, Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'leads' );
-		$leads = $lead_data['data'];
 
-		while ( ( $offset + $limit ) < $lead_data['total'] ) {
-			$offset    = $offset + $limit;
-			$lead_data = $this->idx_api( $api_method . $offset, Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'leads' );
-			$leads     = array_merge( $leads, $lead_data['data'] );
+		if ( array_key_exists( 'data', $lead_data ) && array_key_exists( 'total', $lead_data ) ) {
+			$leads = $lead_data['data'];
+
+			while ( ( $offset + $limit ) < $lead_data['total'] ) {
+				$offset    = $offset + $limit;
+				$lead_data = $this->idx_api( $api_method . $offset, Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'leads' );
+				// If no leads are returned, stop requesting more.
+				if ( empty( $lead_data['data'] ) ) {
+					break;
+				}
+				$leads = array_merge( $leads, $lead_data['data'] );
+			}
 		}
 
 		return $leads;
 	}
 
+	/**
+	 * Get_leads_total function.
+	 *
+	 * @access public
+	 * @return int
+	 */
+	public function get_leads_total() {
+
+		$lead_count = 0;
+
+		$lead_data = $this->idx_api( 'lead?offset=0', Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'leads' );
+
+		if ( array_key_exists( 'total', $lead_data ) && 'integer' === gettype( $lead_data['total'] ) ) {
+			$lead_count = $lead_data['total'];
+		}
+
+		return $lead_count;
+	}
+
+	/**
+	 * Get_recent_active_leads function.
+	 *
+	 * @access public
+	 * @param string $date_type (default: 'subscribeDate').
+	 * @param int    $lead_count (default: 5).
+	 * @return array
+	 */
+	public function get_recent_leads( $date_type = 'subscribeDate', $lead_count = 5 ) {
+		// Get first page of leads.
+		$lead_data = $this->idx_api( 'lead?offset=0&dateType=' . $date_type, Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'leads' );
+
+		if ( ! is_wp_error( $lead_data ) && ! empty( $lead_data['data'] ) ) {
```

Comment:
To avoid nesting of hell with error checking, the pattern of checking for the inverse and returning early helps:
```
if ( errors || emptyData ) {
  return [];
}
// Other code
```
Also applies for line 830